addons/html_builder/static/tests/translation.test.js

```diff
@@ -0,0 +1,113 @@
+    expect(".o-snippets-tabs button:contains('CUSTOMIZE')").toHaveClass("active");
+    expect(".o-snippets-tabs button:contains('CUSTOMIZE')").not.toHaveAttribute("disabled");
+});
+
+test("invisible elements in translate mode", async () => {
+    setupSnippetsMenuForTranslation();
+    await setupWebsiteBuilder(invisiblePopup);
+    expect(".o_we_invisible_el_panel  .o_we_invisible_entry:contains('Popup')").toHaveCount(1);
+});
+
+test("translate text", async () => {
+    setupSnippetsMenuForTranslation();
+    const resultSave = [];
+    onRpc("ir.ui.view", "web_update_field_translations", ({ args }) => {
+        resultSave.push(args[2]["fr_BE"]["sourceSha"]);
+        return true;
+    });
+    const { getEditor } = await setupWebsiteBuilder(getTranslateEditable("Hello"));
+    const editor = getEditor();
+    setContent(editor.editable, getTranslateEditable("H[]ello"));
+    await insertText(editor, "1");
+    await contains(".o-snippets-top-actions button:contains(Save)").click();
+    expect(resultSave.length).toBe(1);
+    expect(resultSave[0]).toBe("H1ello");
+});
+
+test("add text in translate mode do not split", async () => {
+    setupSnippetsMenuForTranslation();
+    const { getEditor } = await setupWebsiteBuilder(getTranslateEditable("Hello"));
+    const editor = getEditor();
+    setContent(editor.editable, getTranslateEditable("Hello[]"));
+    // Event trigger when you press "Enter" => create a new paragraph
+    await manuallyDispatchProgrammaticEvent(editor.editable, "beforeinput", {
+        inputType: "insertParagraph",
+    });
+    await insertText(editor, "1");
+    await animationFrame();
+    expect(":iframe .s_allow_columns p").toHaveCount(1);
+});
+
+test("404 page in translate mode", async () => {
+    patchWithCleanup(WebsiteSystrayItem.prototype, {
+        setup() {
+            websiteServiceInTranslateMode.is404 = () => true;
+            this.websiteService = websiteServiceInTranslateMode;
+        },
+    });
+    await setupWebsiteBuilder(`<h1> Homepage </h1>`, { openEditor: false });
+    await contains(".o-website-btn-custo-primary").click();
+    expect(
+        ".o_popover .o_translate_website_dropdown_item:contains('Translate 404 page')"
+    ).toHaveCount(1);
+    expect(".o_popover .o_edit_website_dropdown_item:contains('Edit 404 page')").toHaveCount(1);
+    expect(".o_popover .o_edit_website_dropdown_item:contains('Create page')").toHaveCount(1);
+});
+
+function getTranslateEditable(inWrap) {
+    return `
+        <div class="container s_allow_columns">
+            <p>
+                <span data-oe-model="ir.ui.view" data-oe-id="526" data-oe-field="arch_db" data-oe-translation-state="to_translate" data-oe-translation-source-sha="sourceSha" class="o_editable">${inWrap}</span>
+            </p>
+        </div>`;
+}
+
+function setupSnippetsMenuForTranslation() {
+    patchWithCleanup(SnippetsMenu.prototype, {
+        setup() {
+            this.props.translation = true;
```

Comment:
I m not a big fan to edit props
It s not possible to patch the parent to be true ?
(If something trigger an update props, you will have a strange issue)